Jenkinsfile

```diff
@@ -0,0 +1,88 @@
+def workpath_linux = "/src/github.com/deis/workflow-cli"
+def keyfile = "tmp/key.json"
+
+def upload_artifacts = { String filepath ->
+	withCredentials([[$class: 'FileBinding', credentialsId: 'e80fd033-dd76-4d96-be79-6c272726fb82', variable: 'GCSKEY']]) {
+		sh "cat \"\${GCSKEY}\" > ${filepath}"
+		sh "make upload-gcs"
+	}
+}
+
+node('windows') {
+	def gopath = pwd() + "\\gopath"
+	env.GOPATH = gopath
+	def workdir = gopath + "\\src\\github.com\\deis\\workflow-cli"
+
+	def pscmd = { String cmd ->
+		"powershell -NoProfile -ExecutionPolicy Bypass -Command \"${cmd}\""
+	}
+
+	dir(workdir) {
+		stage 'Checkout Windows'
+			checkout scm
+		stage 'Install Windows'
+			bat pscmd('.\\make bootstrap')
+		stage 'Test Windows'
+			bat pscmd('.\\make test')
+	}
+}
+
+node('linux') {
+	def gopath = pwd() + "/gopath"
+	env.GOPATH = gopath
+	def workdir = gopath + workpath_linux
+
+	dir(workdir) {
+		stage 'Checkout Linux'
+			checkout scm
+		stage 'Install Linux'
+			sh 'make bootstrap'
+		stage 'Test Linux'
+			sh 'make test'
+	}
+}
+
+stage 'Build and Upload Artifacts'
+
+parallel(
+	revision: {
+		node('linux') {
+			def gopath = pwd() + "/gopath"
+			env.GOPATH = gopath
+			def workdir = gopath + workpath_linux
+
+			dir(workdir) {
+					checkout scm
+					sh 'make bootstrap'
```

Comment:
@Joshua-Anderson do you know if in this latter `parallel` section when we run a new `node('linux')` if we get a different node then what was used above?  Just wondering if we need to run `make bootstrap` again (as it is done above) -- which technically we wouldn't have to if we were using same node...